### DISCUSSION TITLE
Dynamic Registration and Unregistration of Passive Auditing

### DISCUSSION THREAD

**Discussion by cyal1** [PortSwigger/burp-extensions-montoya-api#72] [Q&A]
how to Dynamic Registration and Unregistration of Passive Auditing, without automatically registering auditing upon plugin load.

**Comment by cyal1** [comment]
> how to Dynamic Registration and Unregistration of Passive Auditing, without automatically registering auditing upon plugin load.

I have implemented a button feature that allows users to manually disable passive auditing. Upon clicking the "Disable Passive Auditing" button, the expected behavior is for the passiveAudit function to return an empty AuditResult object, indicating that no auditing should be performed. However, I have observed that when requests are forwarded after disabling passive auditing, they do not go through the auditing process when I re-enable passive auditing, Burp Suite still considers those forwarded requests as audited.

**Reply by Hannah-PortSwigger** [reply]
Hi

Could you provide some more information on what you are trying to achieve and the issues you are encountering?

If you're able to send us a screen recording of the behavior you are experiencing, that would be helpful. You can drop us an email at support@portswigger.net
